include/MidiApple.h

```diff
@@ -0,0 +1,156 @@
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef MIDI_APPLE_H
+#define MIDI_APPLE_H
+
+#include "lmmsconfig.h"
+
+#ifdef LMMS_BUILD_APPLE
+
+#include "MidiClient.h"
+#include "MidiPort.h"
+#include <CoreMIDI/CoreMIDI.h>
+
+
+class QLineEdit;
+
+
+class MidiApple : public QObject, public MidiClient
+{
+    Q_OBJECT
+public:
+	MidiApple();
+	virtual ~MidiApple();
+
+    static QString probeDevice();
+    
+    
+    inline static QString name()
+    {
+        return QT_TRANSLATE_NOOP( "setupWidget", "Apple MIDI" );
+    }
+    
+    
+    
+    virtual void processOutEvent( const MidiEvent & _me,
+                                 const MidiTime & _time,
+                                 const MidiPort * _port );
+    
+    virtual void applyPortMode( MidiPort * _port );
+    virtual void removePort( MidiPort * _port );
+    
+    
+    // list devices as ports
+    virtual QStringList readablePorts() const
+    {
+        return m_inputDevices.keys();
+    }
+    
+    virtual QStringList writablePorts() const
+    {
+        return m_outputDevices.keys();
+    }
+    
+    // return name of port which specified MIDI event came from
+    virtual QString sourcePortName( const MidiEvent & ) const;
+    
+    // (un)subscribe given MidiPort to/from destination-port
+    virtual void subscribeReadablePort( MidiPort * _port,
+                                       const QString & _dest,
+                                       bool _subscribe = true );
+    virtual void subscribeWritablePort( MidiPort * _port,
+                                       const QString & _dest,
+                                       bool _subscribe = true );
+    virtual void connectRPChanged( QObject * _receiver,
+                                  const char * _member )
+    {
+        connect( this, SIGNAL( readablePortsChanged() ),
+                _receiver, _member );
+    }
+    
+    virtual void connectWPChanged( QObject * _receiver,
+                                  const char * _member )
+    {
+        connect( this, SIGNAL( writablePortsChanged() ),
+                _receiver, _member );
+    }
+    
+    virtual bool isRaw() const
+    {
+        return false;
+    }
+    
+    
+    class setupWidget : public MidiClient::setupWidget
+    {
+    public:
+        setupWidget( QWidget * _parent );
+        virtual ~setupWidget();
+        
+        virtual void saveSettings()
+        {
```

Comment:
I'm putting all the curly brackets on a separated line and I'm also trying to follow the coding convention that I can recognize as such from other files, like spaces before and after parentheses.
Just let me know if I'm missing other conventions.

> I believe these curly braces can be removed.

Sorry, I don't understand, which braces can be removed from this code ?